### PR TITLE
Technique 'ambient' node case added

### PR DESCRIPTION
**Pull request by poof86**
The ambient node case is missing from technique/shader parsing

### CODE REVIEW

**Comment by mrdoob** [comment]
We don't support `ambient` in our built-in materials though.

**Comment by poof86** [comment]
But there is support for ambient occlusion textures. I was trying to pick up ambient and emissive textures in my models and set them to the material.aoMap property.

Btw, I am missing a lightMap feature for the Basic/Constant material. I see it was removed with version r71.
To still get the result I want (a detailed diffuse texture blended with a lightMap) I changed the THREE.ShaderChunk[ 'aomap_fragment’ ] chunk with the following:

#ifdef USE_AOMAP
	vec3 ambientOcclusion = texture2D( aoMap, vUv2 ).rgb * aoMapIntensity;
	reflectedLight.indirectDiffuse = min( reflectedLight.indirectDiffuse.rgb * 2.0 * ambientOcclusion.rgb, vec3(1.0) );
	#if defined( USE_ENVMAP ) && defined( STANDARD )
		float dotNV = saturate( dot( geometry.normal, geometry.viewDir ) );
		reflectedLight.indirectSpecular *= computeSpecularOcclusion( dotNV, ambientOcclusion, material.specularRoughness );
	#endif
#endif

Inspired by WestLangley comment:
https://github.com/mrdoob/three.js/pull/5741#issuecomment-66477956 <https://github.com/mrdoob/three.js/pull/5741#issuecomment-66477956>

Maybe ColladaLoader.js can use the ambient texture for material.aoMap and emissive texture for material.lightMap?
Lightmaps on BasicMaterial would use the following shader chunk:

#ifdef USE_LIGHTMAP
	reflectedLight.indirectDiffuse = min( reflectedLight.indirectDiffuse.rgb * 2.0 * texture2D( lightMap, vUv2 ).rgb * lightMapIntensity, vec3(1.0) );
#endif

**Comment by WestLangley** [comment]
@poof86

Lightmaps in three.js are now treated as a source of indirect (ambient) light, and are purely additive to total indirect light.

Ambient occlusion maps in three.js occlude only indirect (ambient) light.

`MeshBasicMaterial` does not respond to lights.

three.js materials no longer have an `ambient` property. The ambient reflectance of the material is assumed to be the same as the diffuse reflectance of the material (a.k.a, the material's `color`).

File: examples/js/loaders/ColladaLoader.js [hunk 1]
```diff
@@ -3791,7 +3792,21 @@ THREE.ColladaLoader = function () {
 		switch ( this.type ) {
 
 			case 'constant':
-
+			
+		                if ( props.emissionMap !== undefined ) {
```
Comment (mrdoob) [inline examples/js/loaders/ColladaLoader.js]: Do you have any example files that use these maps?

**Comment by poof86** [comment]
I understand,

It would be nice if there is an easy way to use the BasicMaterial to blend detailed diffuse textures and prebaked lighting textures. Maybe add a different kind of textureMap property for this technique? The BasicMaterial is ideal for when you want to be as less resource intensive as possible right? So support for this would be great :)

Should I close the pull request?

File: examples/js/loaders/ColladaLoader.js (hunk 1 above)
Comment (poof86) [inline examples/js/loaders/ColladaLoader.js]: I'll make a quick example just a sec

**Comment by WestLangley** [comment]
By the way, `MeshBasicMaterial` does support ambient occlusion maps.

https://github.com/mrdoob/three.js/pull/6588
https://github.com/mrdoob/three.js/pull/6364#commitcomment-11288785

**Comment by mrdoob** [comment]
Now that `MeshBasicMaterial` supports lightMap. Is this PR needed?

**Comment by poof86** [comment]
The lightmap shader uses a different formula, but I will test to see if it has similar results.
Sorry for not posting an example yet, was quite busy and forgot :/

**Comment by Mugen87** [comment]
Old `ColladaLoader` has been removed.

**Pull request closed without merging**